packages/client/src/sync/fetcher/trienodefetcher.ts

```diff
@@ -347,7 +347,7 @@ export class TrieNodeFetcher extends Fetcher<JobTask, Uint8Array[], Uint8Array>
           // add account node data to account trie
           const node = decodeMPTNode(nodeData)
           if (node instanceof LeafMPTNode) {
-            const key = bytesToUnprefixedHex(pathToHexKey(path, node.key(), 'keybyte'))
+            const key = bytesToHex(pathToHexKey(path, node.key(), 'keybyte'))
```

Comment:
We should not do this. The reason for `unprefixedHex` is that the key length is lower (this is an optimization)

Reply:
I recall that this used to be the case, but now I see that we're converting to Uint8Array before encoding, which would actually be invalid (hexToBytes doesn't handle unprefixed). This code should actually throw:

```
            const key = bytesToUnprefixedHex(pathToHexKey(path, node.key(), 'keybyte'))
            ops.push({
              type: 'put',
              key: hexToBytes(key),
              value: node.value(),
            })
            ``` 
            
Since we're doing `key: hexToBytes(key)`

Am I missing something?

Reply:
Hmm you are right, this should definitely error. TrieFetcher is part of snap sync though, maybe @scorbajio knows what's going on here?

I think I was also thinking about this optimization in another context within client. So the unprefixed hex stuff might not be an optimization here. I actually just realized I've not touched this code much :thinking:

Reply:
The `trienodefetcher` store function isn't being unit tested in the ci, it's being tested by simulated test scripts that need to be run manually, so there likely is an error here that's not being handled, but as long as the key is being consistently handled either prefixed or unprefixed, it should work as intended.